### PR TITLE
samrawit / Node week 2

### PR DESCRIPTION
**Pull request by sam39333**

### CODE REVIEW

File: homework/hackyourtemperature/__tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+
+const request = require('supertest');
+const app = require('../server');
+
+test("GET / should return 'hello from backend to frontend!'", async () => {
+  const response = await request(app).get("/");
+  expect(response.status).toBe(200);
+  expect(response.text).toBe("hello from backend to frontend!");
+});
```
Comment (codethenpizza) [inline homework/hackyourtemperature/__tests__/app.test.js:12]: tests are very important and that's good you implement one! 
usually, tests had to cover the most common cases + edge cases, but for let's just implement a few more. 

let's write tests for the next cases: 

- API should respond with `City is not found!` if the city is unknown
- API should respond with a temperature if the city is known

File: homework/hackyourtemperature/server.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      );
+      const data = await response.json();
+  
+      if (data.cod === "404") {
+        res.send({ weatherText: "City is not found!" });
+      } else {
+        const weatherText = `Current temperature in ${city}: ${data.main.temp}°C`;
+        res.send({ weatherText });
+      }
+    } catch (error) {
```
Comment (codethenpizza) [inline homework/hackyourtemperature/server.js:36]: That's great you use `try/catch`! it will allow us to be sure we respond even if `api.openweathermap.org` will be unreachable

File: homework/hackyourtemperature/server.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import fetch from "node-fetch";
+import { keys } from "./sources/keys.js";
+
+
+
+
+
+const app = express();
+const port = 3000;
+app.use(express.json());
```
Comment (codethenpizza) [inline homework/hackyourtemperature/server.js:11]: To have a clear understanding of what's happening in our code it's important to keep everything organized in the files. 
For that let's keep these two lines here in the `server.js`, but define our app with all the routes in the `app.js`.
